federation/pkg/dnsprovider/providers/google/clouddns/clouddns_test.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clouddns
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns/internal/stubs"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+)
+
+func newTestInterface() (dnsprovider.Interface, error) {
+	// Use this to test the real cloud service - insert appropriate project-id.  Default token source will be used.  See
+	// https://github.com/golang/oauth2/blob/master/google/default.go for details.
+	// i, err := dnsprovider.GetDnsProvider(ProviderName, strings.NewReader("\n[global]\nproject-id = federation0-cluster00"))
+	i, err := newFakeInterface() // Use this to stub out the entire cloud service
+	if err != nil {
+		return nil, err
+	}
+	return i, nil
+}
+
+func newFakeInterface() (dnsprovider.Interface, error) {
+	service := stubs.NewService()
+	interface_ := newInterfaceWithStub("", service)
```

Comment:
When wrapping a pre-existing struct that already uses capitalized reserved words as member names (e.g. "Interface"), or when implementing a pre-existing interface in a similar situation (Interface()), I think it's less confusing to retain the name and add an underscore to prevent clashes with reserved words, than it is to invent new names (like istub). I picked this up from the standard library and other libraries.  We have similar situations with 'type'.

I'm inclined to leave it as is.